src/main/java/com/github/rschmitt/dynamicobject/internal/EdnSerialization.java

```diff
@@ -166,7 +166,11 @@ private static Object getReadOptions() {
 
     public static synchronized <T> void registerType(Class<T> type, EdnTranslator<T> translator) {
         EdnTranslatorAdapter<T> adapter = new EdnTranslatorAdapter<>(translator);
-        translatorCache.put(type, adapter);
+        EdnTranslatorAdapter<?> currentAdapter = translatorCache.put(type, adapter);
```

Comment:
Do you want to check for existence in the map first before constructing the adapter?

Reply:
I could, but not sure it's worth it. It reduces some memory allocation (call to the EdnTranslatorAdapter constructor, which sets a field) in favor of adding some extra time (1 extra call to translatorCache#get).